AddressDataType/ConcatenationHelper.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Linq;
+
+namespace InternationalAddress;
+
+public static class ConcatenationHelper
+{
+    public static string JoinNotNullAndNotWhiteSpace(string separator, params string[] items) =>
+        string.Join(separator, items.Where(item => string.IsNullOrWhiteSpace(item)));
```

Comment:
Missing a `!` here? Might want to add the test that would have caught this.

Reply:
Where do you mean? I don't think you need `!` anywhere here.